plugins/builds/index.js

```diff
@@ -27,30 +27,46 @@ const workflowParser = require('screwdriver-workflow-parser');
  * @param  {Boolean}  [config.start]        Whether to start the build or not
  * @return {Promise}
  */
-function createBuild({ jobFactory, buildFactory, eventFactory, pipelineId, jobName, username,
-    scmContext, build, start }) {
-    return Promise.all([
-        eventFactory.get(build.eventId),
-        jobFactory.get({
-            name: jobName,
-            pipelineId
-        })
-    ]).then(([event, job]) => {
-        if (job.state === 'ENABLED') {
-            return buildFactory.create({
-                jobId: job.id,
-                sha: build.sha,
-                parentBuildId: build.id,
-                eventId: build.eventId,
-                username,
-                configPipelineSha: event.configPipelineSha,
-                scmContext,
-                start: start !== false
-            });
-        }
-
-        return null;
+async function createBuild({ jobFactory, buildFactory, eventFactory, pipelineFactory,
+    pipelineId, jobName, username, scmContext, build, start }) {
+    const event = await eventFactory.get(build.eventId);
+    const job = await jobFactory.get({
+        name: jobName,
+        pipelineId
     });
+
+    let prRef = '';
+
+    if (event.type === 'pr') {
+        const scm = buildFactory.scm;
+        const pipeline = await pipelineFactory.get(pipelineId);
+        const token = await pipeline.token;
+        const scmConfig = {
```

Comment:
do we have to query `scm` each time ? Can `prRef` be stored in PR event ?

Reply:
Oh, I think it's better, thanks!

Reply:
@jithin1987 
I fixed to store the `prRef` into the event model.

Please look these.
https://github.com/screwdriver-cd/data-schema/pull/327
https://github.com/screwdriver-cd/models/pull/359